core/ts/MaciState.ts

```diff
@@ -717,6 +728,169 @@ class Poll {
+        }
+
+        const rowStartIndex = this.rbi * batchSize
+        const colStartIndex = this.cbi * batchSize
+        const [ballots1, ballots2] = this.subsidyCalculation(rowStartIndex, colStartIndex)
+
+        const ballotSubrootProof1 = this.ballotTree.genMerkleSubrootPath(rowStartIndex,rowStartIndex + batchSize)
+        const ballotSubrootProof2 = this.ballotTree.genMerkleSubrootPath(colStartIndex,colStartIndex + batchSize)
+
+
+        const newSubsidySalt = genRandomSalt()
+        saltIndex = this.rbi.toString() + "-" + this.cbi.toString()
+        this.subsidySalts[saltIndex] = newSubsidySalt
+        const newSubsidyCommitment = genTallyResultCommitment(this.subsidy, newSubsidySalt, this.treeDepths.voteOptionTreeDepth)
+
+        const packedVals = MaciState.packSubsidySmallVals(
+            this.rbi,
+            this.cbi,
+            this.numSignUps,
+        )
+
+        const inputHash = sha256Hash([
+            packedVals,
+            sbCommitment,
+            currentSubsidyCommitment,
+            newSubsidyCommitment
+        ])
+
+        const circuitInputs = stringifyBigInts({
+            stateRoot,
+            ballotRoot,
+
+            sbSalt,
+            currentSubsidySalt,
+            newSubsidySalt,
+
+            sbCommitment,
+            currentSubsidyCommitment,
+            newSubsidyCommitment,
+            currentSubsidy,
+
+            packedVals, 
+            inputHash,
+
+            ballots1: ballots1.map((x) => x.asCircuitInputs()),
+            ballots2: ballots2.map((x) => x.asCircuitInputs()),
+            votes1: ballots1.map((x) => x.votes),
+            votes2: ballots2.map((x) => x.votes),
+            ballotPathElements1: ballotSubrootProof1.pathElements,
+            ballotPathElements2: ballotSubrootProof2.pathElements,
+        })
+
+        this.increaseSubsidyIndex()
+        return circuitInputs
+    }
+
+    public increaseSubsidyIndex = () => {
+        const batchSize = this.batchSizes.subsidyBatchSize
+        if (this.cbi * batchSize + batchSize < this.ballots.length ) {
+            this.cbi++
+        } else {
+            this.rbi++
+            this.cbi = 0
+        } 
+        return
+    }
+
+    public previousSubsidyIndexToString = ():string => {
+        const batchSize = this.batchSizes.subsidyBatchSize
+        const numBatches = Math.ceil(this.ballots.length/batchSize)
+        let cbi = this.cbi
+        let rbi = this.rbi
+        if (this.cbi === 0 && this.rbi === 0) {
+            return "0-0"
+        }
+        if (this.cbi !== 0 ) {
+            cbi --
+        } else {
+            rbi --
+            cbi = numBatches - 1
+        }
+        return rbi.toString() + "-" + cbi.toString()
+    }
+
+    public coefficientCalculation = (rowBallot: Ballot, colBallot: Ballot): BigInt  => {
+        let sum = BigInt(0)
+        for (let p = 0; p < this.maxValues.maxVoteOptions; p++) {
+            sum += BigInt(rowBallot.votes[p]) * BigInt(colBallot.votes[p])
```

Comment:
```suggestion
            sum += BigInt(rowBallot.votes[p].valueOf()) * BigInt(colBallot.votes[p].valueOf())
```
[The valueOf() method returns the wrapped primitive value of a BigInt object.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/valueOf)